### ISSUE TITLE
AesonException Invalid flag name: \"bytestring--lt-0_10_4\""

### ISSUE THREAD

**Issue by szymonlipinski** [commercialhaskell/stack#4077]
I use stack from Ubuntu 18.04. I've got this error, and I found that I should run `stack upgrade`, so I run it.

```
λ ~/ stack --version
Version 1.5.1 x86_64
Compiled with:
```

and now I get:

```
λ ~/ stack ghc
Downloaded lts-11.13 build plan.    
AesonException "Error in $.packages.cassava.constraints.flags['bytestring--lt-0_10_4']: Invalid flag name: \"bytestring--lt-0_10_4\""
```

**Comment by mgsloan** [comment]
Hmm, so `stack upgrade` installed version 1.5.1?  What does `which stack` say?

I'm guessing that maybe `~/.local/bin` is not on your PATH, and so the newly installed version of stack isn't accessible from the commandline.  So, perhaps add that to your PATH if it's not already there.

If that doesn't work, then perhaps try `stack upgrade --binary-only` or just use the install script:

> curl -sSL https://get.haskellstack.org/ | sh

(see https://docs.haskellstack.org/en/stable/README/)

**Comment by szymonlipinski** [comment]
```λ ~/ stack --version
Version 1.7.1, Git revision 681c800873816c022739ca7ed14755e85a579565 (5807 commits) x86_64 hpack-0.28.2

λ ~/ which stack
/home/szymon/.local/bin/stack

λ ~/ stack upgrade --binary-only
Current Stack version: 1.7.1, available download version: 1.7.1
Skipping binary upgrade, you are already running the most recent version

λ ~/ curl -sSL https://get.haskellstack.org/ | sh
Stack Version 1.7.1 already appears to be installed at:
  /home/szymon/.local/bin/stack
Use 'stack upgrade' or your OS's package manager to upgrade,
or pass '-f' to this script to over-write the existing binary, e.g.:
  curl -sSL https://get.haskellstack.org/ | sh -s - -f


```

**Comment by mgsloan** [comment]
With 1.7.1, you should no longer get the AesonException

**Comment by szymonlipinski** [comment]
I have uninstalled the system's `haskell-stack` and the one installed locally. I've run `curl -sSL https://get.haskellstack.org/ | sh` and now all is fine. Thank you.

**Comment by ursachec** [comment]
Thanks @szymonlipinski, your fix helped on `Debian 9.6`!
